File: fizz_buzz/fizz_buzz.js
"use strict"

function fizz(n)     { return (0 === n % 3) && (0 !== n % 5) }
function buzz(n)     { return (0 !== n % 3) && (0 === n % 5) }
function fizzBuzz(n) { return (0 === n % 3) && (0 === n % 5) }

for(var i=1; i<101; i++) {
  if (fizzBuzz(i)) { console.log(i + ": FizzBuzz!") }
  else if (fizz(i)) { console.log(i + ": Fizz!") }
  else if (buzz(i)) { console.log(i + ": Buzz!") }
  else { console.log(i) }
}

File: fizz_buzz/fizz_buzz.rb
def fizz(n)
  (0 == n % 3) && (0 != n % 5)
end

def buzz(n)
  (0 != n % 3) && (0 == n % 5)
end

def fizz_buzz(n)
  (0 == n % 3) && (0 == n % 5)
end

(1..100).each do |i|
  if fizz(i)
    puts "#{i}: Fizz!"
  elsif buzz(i)
    puts "#{i}: Buzz!"
  elsif fizz_buzz(i)
    puts "#{i}: FizzBuzz!"
  else
    puts "#{i}"
  end
end
